### PR TITLE
Got web to work with different timezones but still working on mobile

### PR DESCRIPTION
**Pull request by swanka**

### CODE REVIEW

File: packages/ui/src/DateTimeActionSheet.tsx [hunk 1]
```diff
@@ -180,47 +181,60 @@ export function DateTimeActionSheet({
...
   let m;
   if (value) {
-    m = dayjs(value, ["YYYY", "YYYY-MM-DD", "HH:mm", "hh:mm A"]);
+    m = dayjs(value, [
+      "YYYY",
+      "YYYY-MM-DD",
+      "HH:mm",
+      "hh:mm A",
+      transformValue?.options?.transformFormat,
+    ]).tz(transformValue?.options?.timezone);
```
Comment (kjkurtz) [inline packages/ui/src/DateTimeActionSheet.tsx:197]: if you pass in undefined here due to the optionals, does it default to local?

**Review by kjkurtz — APPROVED** [review]

File: packages/ui/src/DateTimeActionSheet.tsx (hunk 1 above)
Comment (swanka) [reply packages/ui/src/DateTimeActionSheet.tsx:197]: I must have missed this question but yes. Undefined returns to the default local display

**Comment by joshgachnang** [comment]
@swanka this should be good now!

**Review by hchamorro — COMMENTED** [review]
Not sure how the demo works, I don't think it works right on master. 
I did some testing against flourish repo and i'll post some preliminary bugs(?)

**Comment by hchamorro** [comment]
> Not sure how the demo works, I don't think it works right on master. I did some testing against flourish repo and i'll post some preliminary bugs(?)

selected date not highlighted in both web and native. video for web below

https://github.com/FlourishHealth/ferns-ui/assets/53444258/8625c2b4-bacc-457b-a3b1-fb2cf54d5253

**Comment by hchamorro** [comment]
> Not sure how the demo works, I don't think it works right on master. I did some testing against flourish repo and i'll post some preliminary bugs(?)

timezone selection showing for date selection both web and mobile. picture from web below:
Screenshot 2024-03-01 at 12 19 48 PM

**Comment by hchamorro** [comment]
> Not sure how the demo works, I don't think it works right on master. I did some testing against flourish repo and i'll post some preliminary bugs(?)

not sure what updates need to be adjusted for flourish --
In web and/or app
When selecting **+ Scheduled Item**
errors: Invalid Date
Screenshot 2024-03-01 at 12 40 16 PM
Screenshot 2024-03-01 at 12 20 33 PM

**Comment by hchamorro** [comment]
> Not sure how the demo works, I don't think it works right on master. I did some testing against flourish repo and i'll post some preliminary bugs(?) Nit: rename PR, remove 'still working on mobile'

not sure if this is something being worked out with this pr. When selecting edit date, it still highlights the day before (UTC shtuff)

File: packages/ui/src/DateTimeActionSheet.tsx [hunk 2]
```diff
@@ -1,18 +1,20 @@
 import {Picker} from "@react-native-picker/picker";
+import {getCalendars} from "expo-localization";
```
Comment (kjkurtz) [inline packages/ui/src/DateTimeActionSheet.tsx:2]: Nice! I didn't know this existed.

**Review by kjkurtz — APPROVED** [review]
This seems solid to me! I tested on web, iOS, and android. I think most of the issues that Hannah saw are due to updates that need to be made within Flourish once this is merged.

**Comment by kjkurtz** [comment]
| timezone selection showing for date selection both web and mobile.
I did notice this one as well.

File: .vscode/settings.json [hunk 3]
```diff
@@ -1,10 +1,10 @@
 {
   "editor.codeActionsOnSave": {
-    "source.fixAll.eslint": true
+    "source.fixAll.eslint": "explicit"
```
Comment (joshgachnang) [inline .vscode/settings.json:3]: Why the changes?

Comment (kjkurtz) [reply .vscode/settings.json:3]: VSCode/eslint does this by default in the newest version, so this is right.
